### DISCUSSION TITLE
Extra GPIO on JTAG mode

### DISCUSSION THREAD

**Discussion by Bnjamn** [eblot/pyftdi#330] [General]
Hello,
I'm very happy about this FTDI package and I've been using it for several electronics project with FTDI chips.
Right now, I will need to run a JTAG link, and I've connected extra GPIO on the FTDI bus.
Is there a way to drive them like the way it does for spi or i2c ?

Thanks in advance for your helps,
Best,
Benjamin.

**Comment by eblot** [comment]
I'm not sure to get the question.

There is a JTAG module in PyFtdi, but I'm not sure to understand your needs. Can you elaborate?

**Reply by Bnjamn** [reply]
Thanks for your reply.
No probs, let me explain it a bit more :
It seems that the JTAG link will not take all the I/Os on a FTDI bus (ADBUS, BDBUS, ...).
So I'm just asking about the possibility to use the spare I/Os that will remain on the BUS.

**Comment by eblot** [comment]
Ok, I got it.

Technically, there would be no issue to manage non-JTAG pins as regular IOs.

Unfortunately I have not worked on the JTAG part of PyFtdi for years, and I do not think this feature is implemented. Patches are welcome though. This feature exists for other MPSSE usage (I2C,SPI), so it should not be a big trouble to implement. 
I have no spare time to dedicate to this feature myself though...

**Comment by feliggs** [comment]
@Bnjamn I have a question regarding the GPIOs and how to control them. Is there a way I can get in touch with you to discuss it further? Any help would be greatly appreciated. Thank you!

**Comment by MBrent65** [comment]
Thank you very much for your work, it will help me a lot with a gws module. 
Well I don't want to speak too soon so as not to attract the bad eye.
That's already a great help, 
Thank you again.
